File: mobile-android/app/src/main/java/com/lfork/blogsystem/utils/ToastUtil.java
package com.lfork.blogsystem.utils;

import android.content.Context;
import android.os.Looper;
import android.widget.Toast;

/**
 *
 * @author 98620
 * @date 2018/4/14
 */


public class ToastUtil {

    private final static String MAIN_THREAD = "main";

    public static void showLong(Context context, String content) {
        if (!MAIN_THREAD.equals(Thread.currentThread().getName())) {
            if (Looper.myLooper() == null) {
                Looper.prepare();
            }
        }
        Toast.makeText(context, content, Toast.LENGTH_LONG).show();
    }

    public static void showShort(Context context, String content) {
        //不能在非主线程里面直接Toast   Can't create handler inside thread that has not called Looper.prepare()
        if (!MAIN_THREAD.equals(Thread.currentThread().getName())) {
            if (Looper.myLooper() == null) {
                Looper.prepare();
            }
        }
        Toast.makeText(context, content, Toast.LENGTH_SHORT).show();
    }
}

File: mobile-android/app/src/main/java/com/lfork/blogsystem/utils/ToastUtil.kt
package com.lfork.blogsystem.utils

import android.content.Context
import android.os.Looper
import android.util.Log
import android.widget.Toast

/**
 *
 * @author 98620
 * @date 2018/4/14
 */


object ToastUtil {

    private val MAIN_THREAD = "main"

    private val TAG = "ToastUtil"

    fun showLong(context: Context?, content: String?) {
        if (MAIN_THREAD != Thread.currentThread().name) {
            if (Looper.myLooper() == null) {
                Looper.prepare()
            }
        }

        if(context == null) {
            Log.w(TAG, "Context is null")
        }
        Toast.makeText(context, content, Toast.LENGTH_LONG).show()
    }

    fun showShort(context: Context?, content: String?) {
        //不能在非主线程里面直接Toast   Can't create handler inside thread that has not called Looper.prepare()
        if (MAIN_THREAD != Thread.currentThread().name) {
            if (Looper.myLooper() == null) {
                Looper.prepare()
            }
        }
        if(context == null) {
            Log.w(TAG, "Context is null")
        }
        Toast.makeText(context, content, Toast.LENGTH_SHORT).show()
    }

}
